### DISCUSSION TITLE
Scrollable page via sprite?

### DISCUSSION THREAD

**Discussion by rradomir** [Bodmer/TFT_eSPI#2434] [Q&A - Sprites]
When I only "discovered" sprite functionality, instantly thought about this.
It is good approach?
My goal is to draw page at once, and then print only visible part and just scroll it. Then I can change values of bars. For now they are mockups of course. Please review, if you have time for that :D
https://github.com/rradomir/Page-Scroll-TFT_eSPI

I thinking of updating only one changed bar. I think it is possible to update sprite and push to screen only changed part of it. Im only need to calculate offsets.  Of course when I scroll page, all of it must be redrawed.
Handle length is calculated at start, so changing length of page should extend/shrink also scrollbar.
Im surprised that redrawing page isn't that laggy but im quickly running out of RAM :D

**Comment by Bodmer** [comment]
This is a reasonable approach for small screen areas, less practical for large areas due to the memory consumed.

**Comment by rradomir** [comment]
Yes, indeed.

I have two questions associated with it.
1) it is possible to declare space for sprites in code, like variables? As i assume, creating sprite allocating space dynamically. I already got a freeze, when ran out of memory once, when I tried 16 bit sprite :) it could be easier to spot, if I get info about space while compiling.
2) I wonder if this would have sense, to save & restore sprite in temporary file on sd card? It would cause impact for rendering speed, but in other hand it could be then almost infinite (of "overall area")?
